src/components/site-card/editor-launcher.tsx

```diff
@@ -4,20 +4,27 @@ import { IconButton, Tooltip } from "gatsby-interface"
 import { useCallback, useMemo } from "react"
 import openInEditor from "open-in-editor"
 import { editorIcons, CodeEditor, editorLabels } from "../../util/editors"
+import { trackEvent } from "../../util/telemetry"
 
 export interface IProps {
   path: string
   editor?: CodeEditor
+  siteHash?: string
 }
 
-export function EditorLauncher({ path, editor = `code` }: IProps): JSX.Element {
+export function EditorLauncher({
+  path,
+  editor = `code`,
+  siteHash,
+}: IProps): JSX.Element {
   const editorInstance = useMemo(() => openInEditor.configure({ editor }), [
     editor,
   ])
 
   const openEditor = useCallback((): void => {
+    trackEvent(`LAUNCH_EDITOR`, { name: editor, siteHash })
     editorInstance.open(path)
-  }, [editorInstance])
+  }, [editorInstance, siteHash])
```

Comment:
A personal suggestion: I would add `editor` as a dependency and make `editorInstance` a simple function instead of memo-ed one (I prefer having more data and less functions as dependency in this array, but as I mentioned, it's a personal preference)

Reply:
Good catch with the dependency, but`editorInstance` is memoized because it's expensive to create. It does a load of setup and checks.

Reply:
Actually, moving it inside useCallback has the same effect, so that's cleaner

Reply:
Oh, no wait that's no good as it recreates it on every click! Never mind